src/rust/catnap/futures/pop.rs

```diff
@@ -78,34 +81,40 @@ impl Future for PopFuture {
     /// Polls the target [PopFuture].
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
         let self_: &mut PopFuture = self.get_mut();
-        let mut bytes: [u8; POP_SIZE] = [0; POP_SIZE];
-        match socket::recvfrom::<SockaddrStorage>(self_.fd, &mut bytes[..]) {
+        match unsafe {
+            let mut addrlen: libc::socklen_t = 0;
```

Comment:
Another one.

Reply:
You are correct. I have fixed this. Thanks for checking this.